docs_app/src/app/shared/state.service.ts

```diff
@@ -0,0 +1,171 @@
+    op1: OperatorFunction<T, A>,
+    op2: OperatorFunction<A, B>,
+    op3: OperatorFunction<B, C>,
+    op4: OperatorFunction<C, D>,
+  ): Observable<D>;
+  select<A = T, B = A, C = B, D = C, E = D>(
+    op1: OperatorFunction<T, A>,
+    op2: OperatorFunction<A, B>,
+    op3: OperatorFunction<B, C>,
+    op4: OperatorFunction<C, D>,
+    op5: OperatorFunction<D, E>,
+  ): Observable<E>;
+  // ================================
+  select<K1 extends keyof T>(k1: K1): Observable<T[K1]>;
+  select<K1 extends keyof T,
+    K2 extends keyof T[K1]>(k1: K1, k2: K2): Observable<T[K1][K2]>;
+  select<K1 extends keyof T,
+    K2 extends keyof T[K1],
+    K3 extends keyof T[K1][K2]>(k1: K1, k2: K2, k3: K3): Observable<T[K1][K2][K3]>;
+  select<K1 extends keyof T,
+    K2 extends keyof T[K1],
+    K3 extends keyof T[K1][K2],
+    K4 extends keyof T[K1][K2][K3]>(k1: K1, k2: K2, k3: K3, k4: K4): Observable<T[K1][K2][K3][K4]>;
+  select<K1 extends keyof T,
+    K2 extends keyof T[K1],
+    K3 extends keyof T[K1][K2],
+    K4 extends keyof T[K1][K2][K3],
+    K5 extends keyof T[K1][K2][K3][K4]>(k1: K1, k2: K2, k3: K3, k4: K4, k5: K5): Observable<T[K1][K2][K3][K4][K5]>;
+  select<K1 extends keyof T,
+    K2 extends keyof T[K1],
+    K3 extends keyof T[K1][K2],
+    K4 extends keyof T[K1][K2][K3],
+    K5 extends keyof T[K1][K2][K3][K4],
+    K6 extends keyof T[K1][K2][K3][K4][K5]>(k1: K1, k2: K2, k3: K3, k4: K4, k5: K5, k6: K6): Observable<T[K1][K2][K3][K4][K5][K6]>;
+  // ===========================
+  select(...opOrMapFn: any[]): Observable<any> {
+    if (!opOrMapFn || opOrMapFn.length === 0) {
+      return this.state$
+        .pipe(
+          stateful()
+        );
+    } else if (!this.isStringArray(opOrMapFn)) {
+      const path = (opOrMapFn as any) as string[];
+      return this.state$.pipe(
+        pluck(...path),
+        stateful()
+      );
+    } else if (this.isOperateFnArray(opOrMapFn)) {
+      const oprs = opOrMapFn as OperatorFunction<T, any>[];
+      return this.state$.pipe(
+        pipeFromArray(oprs),
+        stateful()
+      );
+    }
+
+    throw new Error('Wrong params passed' + JSON.stringify(opOrMapFn));
+  }
+
+  holdEffect<S>(observableWithSideEffect: Observable<S>): void;
+  holdEffect<S>(obsOrObsWithSideEffect: Observable<S>, sideEffectFn?: (arg: S) => void): void {
+    if (sideEffectFn) {
+      this.effectSubject.next(obsOrObsWithSideEffect.pipe(tap(sideEffectFn)));
+    }
+    this.effectSubject.next(obsOrObsWithSideEffect);
+  }
+
+  teardown(): void {
+    this.subscription.unsubscribe();
+  }
+
+  private isOperateFnArray(op: any[]): op is OperatorFunction<T, any>[] {
+    return op.every((i: any) => typeof i !== 'string');
+  }
+
+  private isStringArray(op: any[]): op is string[] {
+    return op.every((i: any) => typeof i !== 'string');
+  }
+
+  ngOnDestroy(): void {
+    this.teardown();
+  }
+
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GlobalState<T> extends State<T> {
```

Comment:
Some of the stuff in this file is not used at all, please clean up all the stuff not is not used

Reply:
done

Reply:
done